test/integration/elasticquota_controller_test.go

```diff
@@ -0,0 +1,355 @@
+					Max(MakeResourceList().CPU(100).Mem(1000).Obj()).Obj(),
+				MakeEQ("ns2", "t2-eq2").
+					Min(MakeResourceList().CPU(100).Mem(1000).Obj()).
+					Max(MakeResourceList().CPU(100).Mem(1000).Obj()).Obj(),
+			},
+			nowPods: []*v1.Pod{
+				MakePod("ns1", "t2-p1").Phase(v1.PodRunning).Node("fake-node").
+					Container(MakeResourceList().CPU(10).Mem(20).Obj()).Obj(),
+				MakePod("ns1", "t2-p2").Phase(v1.PodRunning).Node("fake-node").
+					Container(MakeResourceList().CPU(10).Mem(10).Obj()).Obj(),
+				MakePod("ns1", "t2-p3").Phase(v1.PodRunning).Node("fake-node").
+					Container(MakeResourceList().CPU(10).Mem(10).Obj()).Obj(),
+				MakePod("ns2", "t2-p4").Phase(v1.PodRunning).Node("fake-node").
+					Container(MakeResourceList().CPU(10).Mem(10).Obj()).Obj(),
+			},
+			nowEQUsed: []*v1alpha1.ElasticQuota{
+				MakeEQ("ns1", "t2-eq1").
+					Used(MakeResourceList().CPU(30).Mem(40).Obj()).Obj(),
+				MakeEQ("ns2", "t2-eq2").
+					Used(MakeResourceList().CPU(10).Mem(10).Obj()).Obj(),
+			},
+			nextPods: []*v1.Pod{
+				MakePod("ns1", "t2-p1").Phase(v1.PodSucceeded).Obj(),
+				MakePod("ns1", "t2-p3").Phase(v1.PodFailed).Obj(),
+			},
+			nextEQUsed: []*v1alpha1.ElasticQuota{
+				MakeEQ("ns1", "t2-eq1").
+					Used(MakeResourceList().CPU(10).Mem(10).Obj()).Obj(),
+				MakeEQ("ns2", "t2-eq2").
+					Used(MakeResourceList().CPU(10).Mem(10).Obj()).Obj(),
+			},
+		},
+		{
+			name: "Different resource between max and min",
+			elasticQuotas: []*v1alpha1.ElasticQuota{
+				MakeEQ("ns1", "t3-eq1").
+					Min(MakeResourceList().Mem(1000).Obj()).
+					Max(MakeResourceList().CPU(100).Obj()).Obj(),
+			},
+			nowPods: []*v1.Pod{
+				MakePod("ns1", "t3-p1").Node("none").
+					Container(MakeResourceList().CPU(10).Mem(20).Obj()).Obj(),
+			},
+			nowEQUsed: []*v1alpha1.ElasticQuota{
+				MakeEQ("ns1", "t3-eq1").
+					Used(MakeResourceList().CPU(0).Mem(0).Obj()).Obj(),
+			},
+			nextPods: []*v1.Pod{
+				MakePod("ns1", "t3-p1").Phase(v1.PodRunning).Node("fake-node").
+					Container(MakeResourceList().CPU(10).Mem(20).Obj()).Obj(),
+			},
+			nextEQUsed: []*v1alpha1.ElasticQuota{
+				MakeEQ("ns1", "t3-eq1").
+					Used(MakeResourceList().CPU(10).Mem(20).Obj()).Obj(),
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			defer cleanupElasticQuotas(ctx, extClient, tt.elasticQuotas)
+			defer testutil.CleanupPods(cs, t, tt.nowPods)
+			defer testutil.CleanupPods(cs, t, tt.nextPods)
+			// create elastic quota
+			if err := createElasticQuotas(ctx, extClient, tt.elasticQuotas); err != nil {
+				t.Fatal(err)
+			}
+
+			// create now pod and update status
+			for _, pod := range tt.nowPods {
+				_, err := cs.CoreV1().Pods(pod.Namespace).Create(testCtx.Ctx, pod, metav1.CreateOptions{})
+				if err != nil {
+					t.Fatalf("Failed to create Pod %q: %v", pod.Name, err)
+				}
+				if pod.Status.Phase == v1.PodRunning {
+					_, err = cs.CoreV1().Pods(pod.Namespace).UpdateStatus(testCtx.Ctx, pod, metav1.UpdateOptions{})
```

Comment:
Why we need to update the status here?

Reply:
Because the controller only counts pods in the running state

Reply:
Do you mean when the `existingPods` were created at L274, as scheduler is not involved, they're in pending instead of running state?

Reply:
Not exactly, the main reason is that there is no kubelet to update the pod status